fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindSlotReference.java

```diff
@@ -156,7 +179,12 @@ public Expression visitUnboundAlias(UnboundAlias unboundAlias, Void context) {
         }
 
         @Override
-        public Slot visitUnboundSlot(UnboundSlot unboundSlot, Void context) {
+        public Slot visitUnboundSlot(UnboundSlot unboundSlot, PlannerContext context) {
+            List<Slot> tmpBound = bindSlot(unboundSlot, getScope().getSlots());
+            boolean hasCorrelate = false;
```

Comment:
this position should has a variable named as `boolean foundInThisScope = !tmpBound.isEmpty()`;
and if foundInThisScope is false, then set `boolean foundInOuterScope = true`

and then is the switch statement, you can declare `boolean hasCorrelate = foundInOuterScope` or use `foundInOuterScope` directly